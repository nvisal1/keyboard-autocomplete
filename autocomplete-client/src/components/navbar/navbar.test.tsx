import React from 'react';
import renderer from 'react-test-renderer';

import Navbar, { NavbarProps } from './navbar';

describe('When the Navbar component is created', () => {
    it('should render correctly', () => {
        const requiredProps = getRequiredProps();
        const currentSnapshot = renderer.create(<Navbar { ...requiredProps }></Navbar>).toJSON();

        expect(currentSnapshot).toMatchSnapshot();
    });
});

function getRequiredProps(): NavbarProps {
   const handleClick = (_: string) => {}; 

   const selectedOption = 'test_option';

   const options = ['test_option'];

   return { handleClick, selectedOption, options };
}