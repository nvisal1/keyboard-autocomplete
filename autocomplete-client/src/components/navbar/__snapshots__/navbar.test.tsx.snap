// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When the Navbar component is created should render correctly 1`] = `
<div
  className="navbar"
>
  <div
    className="navbar__options-container"
  >
    <div
      className="option-container__text"
      onClick={[Function]}
      style={
        Object {
          "color": "#BE1E2D",
        }
      }
    >
      test_option
    </div>
  </div>
</div>
`;
